src/app/core/services/schedule/schedule-factory.service.ts

```diff
@@ -75,6 +75,6 @@ export class ScheduleFactoryService extends ScheduleService {
   }
 
   updateTaskToComplete(updatedTask): Promise<any> {
-    return this.scheduleService.updateTaskToComplete(updatedTask)
```

Comment:
This should be done for both types of scheduleservices. In local one the updateTaskToComplete function will also handle deleting of notifications for completed tasks.